partner_external_maps/partner_external_maps.py

```diff
@@ -0,0 +1,197 @@
+    lat_lon_url = fields.Char(
+        string='URL that uses latitude and longitude',
+        help="In this URL, {LATITUDE} and {LONGITUDE} will be replaced by "
+        "the latitude and longitude (requires the module 'base_geolocalize')")
+    route_address_url = fields.Char(
+        string='Route URL that uses the addresses',
+        help="In this URL, {START_ADDRESS} and {DEST_ADDRESS} will be "
+        "replaced by the start and destination addresses.")
+    route_lat_lon_url = fields.Char(
+        string='Route URL that uses latitude and longitude',
+        help="In this URL, {START_LATITUDE}, {START_LONGITUDE}, "
+        "{DEST_LATITUDE} and {DEST_LONGITUDE} will be replaced by the "
+        "latitude and longitude of the start and destination adresses "
+        "(requires the module 'base_geolocalize').")
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    # begin with context_ to allow user to change it by himself
+    context_map_website_id = fields.Many2one(
+        'map.website', string='Map Website',
+        domain=[
+            '|', ('address_url', '!=', False), ('lat_lon_url', '!=', False)])
+    # We want to give the possibility to the user to have one map provider for
+    # regular maps and another one for routing
+    context_route_map_website_id = fields.Many2one(
+        'map.website', string='Route Map Website',
+        domain=[
+            '|',
+            ('route_address_url', '!=', False),
+            ('route_lat_lon_url', '!=', False)],
+        help="Map provided used when you click on the car icon on the partner "
+        "form to display an itinerary.")
+    context_route_start_partner_id = fields.Many2one(
+        'res.partner', string='Start Address for Route Map')
+
+    @api.model
+    def _default_map_settings(self):
+        '''Method called from post-install script
+        I can't use a default method on the field, because it would be executed
+        before loading map_website_data.xml, so it would not be able to set a
+        value'''
+        users = self.env['res.users'].search([])
+        map_website = self.env['map.website'].search([
+            '|', ('address_url', '!=', False), ('lat_lon_url', '!=', False)],
+            limit=1)
+        map_route_website = self.env['map.website'].search([
+            '|',
+            ('route_address_url', '!=', False),
+            ('route_lat_lon_url', '!=', False)], limit=1)
+        logger.info('Updating user settings for maps...')
+        for user in users:
+            user.write({
+                'context_map_website_id': map_website.id or False,
+                'context_route_map_website_id': map_route_website.id or False,
+                'context_route_start_partner_id': user.partner_id.id or False,
+                })
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.model
+    def _address_as_string(self):
+        addr = u''
+        if self.street:
```

Comment:
Isn't better to add all existing fields to a list and build the string with the `' '.join()` expression?